File: design-patterns/go/chain_of_responsibility/interview/phone_interview.go
// phone_interview.go

package main

import "fmt"

type PhoneInterview struct {
	nextInterview Interview
}

func NewPhoneInterview(nextInterview Interview) (phoneInterview *PhoneInterview) {
	phoneInterview = &PhoneInterview{}
	phoneInterview.nextInterview = nextInterview
	return
}

func (phoneInterview *PhoneInterview) Execute() {
	// Ask all questions for phone interview
	// Perform any other action required for phone interview
	fmt.Println("Ask phone interview questions")

	// Execute the next interview set while creating new struct
	if phoneInterview.nextInterview != nil {
		phoneInterview.nextInterview.Execute()
	}
}

File: design-patterns/ts/chain-of-responsibility/interview/phone-interview.ts
// phone-interview.ts

import Interview from "./interview";

class PhoneInterview extends Interview {
    constructor(nextInterview: Interview | null) {
        super(nextInterview);
    }

    execute(): void {
        // Ask all questions for phone interview
        // Perform any other action required for phone interview
        console.log("Ask phone interview questions")

        // Execute the next interview set while creating new struct
        if (this.nextInterview) {
            this.nextInterview.execute();
        }
    }

}

export default PhoneInterview;